### ISSUE TITLE
Can't "Prepare Interface"

### ISSUE THREAD

**Issue by WavesOfSalt** [jellyfin/jellyfin-tizen#128]
The guide's command; ```JELLYFIN_WEB_DIR=../jellyfin-web/dist npm ci --no-audit```, does not work under Windows. Apparently Jeppe Beier managed to get things working using ```$env:JELLYFIN_WEB_DIR='../jellyfin-web/dist npm ci --no-audit```, but npm ci --no-audit does not work.
```Using jellyfin-web from E:\Tizen\gitted\jellyfin-web\dist npm ci --no-audit
[19:30:50] Using gulpfile E:\Tizen\gitted\jellyfin-tizen\gulpfile.js
[19:30:50] Starting 'default'...
[19:30:50] Starting 'clean'...
[19:30:50] Finished 'clean' after 12 ms
[19:30:50] Starting 'searchFonts'...
[19:30:50] Finished 'searchFonts' after 17 ms
[19:30:50] Starting 'copy'...
[19:30:50] Starting 'modifyIndex'...
[19:30:50] Finished 'copy' after 9.28 ms
[19:30:50] 'modifyIndex' errored after 13 ms
[19:30:50] Error: File not found with singular glob: E:\Tizen\gitted\jellyfin-web\dist npm ci --no-audit/index.html (if this was purposeful, use `allowEmpty` option)
    at Glob.<anonymous> (E:\Tizen\gitted\jellyfin-tizen\node_modules\glob-stream\readable.js:84:17)
    at Object.onceWrapper (node:events:642:26)
    at Glob.emit (node:events:527:28)
    at Glob.emit (node:domain:475:12)
    at Glob._finish (E:\Tizen\gitted\jellyfin-tizen\node_modules\glob\glob.js:194:8)
    at done (E:\Tizen\gitted\jellyfin-tizen\node_modules\glob\glob.js:179:14)
    at Glob._processSimple2 (E:\Tizen\gitted\jellyfin-tizen\node_modules\glob\glob.js:685:12)
    at E:\Tizen\gitted\jellyfin-tizen\node_modules\glob\glob.js:673:10
    at Glob._stat2 (E:\Tizen\gitted\jellyfin-tizen\node_modules\glob\glob.js:769:12)
    at lstatcb_ (E:\Tizen\gitted\jellyfin-tizen\node_modules\glob\glob.js:761:12)
[19:30:50] 'default' errored after 51 ms
npm ERR! code 1
npm ERR! path E:\Tizen\gitted\jellyfin-tizen
npm ERR! command failed
npm ERR! command C:\WINDOWS\system32\cmd.exe /d /s /c gulp```

**Comment by Gurkward** [comment]
As seen by your error `[19:30:50] Error: File not found with singular glob: E:\Tizen\gitted\jellyfin-web\dist npm ci --no-audit/index.html` you've accidentally messed up the path
You're supposed to run `set JELLYFIN_WEB_DIR=../jellyfin-web/dist` and `npm ci --no-audit` separately

**Comment by WavesOfSalt** [comment]
I don't know how I messed up the path. I didn't change it and,  after all, the guide on https://github.com/jellyfin/jellyfin-tizen shows these with one line (https://archive.is/IncoF),
![image](https://user-images.githubusercontent.com/65592397/236641651-44d889a8-dfa0-4d40-85b1-91bca31d229e.png)
And unless the command is expecting an input and can just function with nothing, commands are read from left to right. So it shouldn't make a difference.

But I gave up on this since somebody else was building and uploading some builds, but the tv was just not accepting any installs. Apparently, for those Tizen TVs that have some kind of web browser, just browsing to the IP of the Jellyfin server works

Having said that, I tried to give this another go. Compiling gives random errors, but eventually it seemed building Jellyfin Web worked. Then I went back to the step of preparing the interface. Something additional happened. And this was from running npm ci --no-audit separately.
[15:16:33] Error: File not found with singular glob: E:\Tizen\gitted\jellyfin-tizen\node_modules\jellyfin-web\dist/index.html (if this was purposeful, use `allowEmpty` option)
    at Glob.<anonymous> (E:\Tizen\gitted\jellyfin-tizen\node_modules\glob-stream\readable.js:84:17)
    at Object.onceWrapper (node:events:642:26)
    at Glob.emit (node:events:527:28)
    at Glob.emit (node:domain:475:12)
    at Glob._finish (E:\Tizen\gitted\jellyfin-tizen\node_modules\glob\glob.js:194:8)
    at done (E:\Tizen\gitted\jellyfin-tizen\node_modules\glob\glob.js:179:14)
    at Glob._processSimple2 (E:\Tizen\gitted\jellyfin-tizen\node_modules\glob\glob.js:685:12)
    at E:\Tizen\gitted\jellyfin-tizen\node_modules\glob\glob.js:673:10
    at Glob._stat2 (E:\Tizen\gitted\jellyfin-tizen\node_modules\glob\glob.js:769:12)
    at lstatcb_ (E:\Tizen\gitted\jellyfin-tizen\node_modules\glob\glob.js:761:12)
[15:16:33] 'default' errored after 45 ms
npm ERR! code 1
npm ERR! path E:\Tizen\gitted\jellyfin-tizen
npm ERR! command failed
npm ERR! command C:\WINDOWS\system32\cmd.exe /d /s /c gulp

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\****\AppData\Local\npm-cache\_logs\2023-05-06T19_15_57_611Z-debug-0.log

**Comment by dmitrylyzo** [comment]
> And this was from running npm ci --no-audit separately.
> [15:16:33] Error: File not found with singular glob: E:\Tizen\gitted\jellyfin-tizen\node_modules\jellyfin-web\dist/index.html (if this was purposeful, use `allowEmpty` option)

`JELLYFIN_WEB_DIR` environment variable is not set.
You should run both commands if you have closed the command line window.
https://github.com/jellyfin/jellyfin-tizen/wiki/Build-on-Windows#prepare-interface
_There are commands for Windows Command Prompt and PowerShell._

**Comment by WavesOfSalt** [comment]
Sorry if I sound rude, but are you sure? 
![image](https://github.com/jellyfin/jellyfin-tizen/assets/65592397/de1d739d-6087-441b-a1d4-7c8a1faf5713)
![image](https://github.com/jellyfin/jellyfin-tizen/assets/65592397/c75990e2-ea86-4cea-bd37-1244b31cd0ec)
![image](https://github.com/jellyfin/jellyfin-tizen/assets/65592397/03f9dd3b-baa1-49db-a337-5ac7d190b3cd)
![image](https://github.com/jellyfin/jellyfin-tizen/assets/65592397/55f7d68d-00d3-4a41-845f-b6c28cb06aba)
![image](https://github.com/jellyfin/jellyfin-tizen/assets/65592397/ee0e6726-ebe3-4320-8bc1-d386262a6bfe)


```
PS E:\Tizen\gitted> git clone -b release-10.8.z https://github.com/jellyfin/jellyfin-web.git
Cloning into 'jellyfin-web'...
remote: Enumerating objects: 218930, done.
remote: Counting objects: 100% (18384/18384), done.
remote: Compressing objects: 100% (814/814), done.
remote: Total 218930 (delta 17824), reused 17783 (delta 17569), pack-reused 200546
Receiving objects: 100% (218930/218930), 120.97 MiB | 8.37 MiB/s, done.

Resolving deltas: 100% (151517/151517), done.
PS E:\Tizen\gitted> git clone https://github.com/jellyfin/jellyfin-tizen.git
Cloning into 'jellyfin-tizen'...
remote: Enumerating objects: 222, done.
remote: Counting objects: 100% (116/116), done.
remote: Compressing objects: 100% (70/70), done.
Receiving objects: 100% (222/222), 134.14 KiB | 1.86 MiB/s, done.06

Resolving deltas: 100% (107/107), done.
PS E:\Tizen\gitted> cd jellyfin-web
>> $env:SKIP_PREPARE=1
>> npm ci --no-audit
>> npm run build:production
npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.
npm WARN old lockfile
npm WARN old lockfile The package-lock.json file was created with an old version of npm,
npm WARN old lockfile so supplemental metadata must be fetched from the registry.
npm WARN old lockfile
npm WARN old lockfile This is a one-time fix-up, please be patient...
npm WARN old lockfile
npm WARN deprecated @types/vfile-message@2.0.0: This is a stub types definition. vfile-message provides its own type definitions, so you do not need this installed.
npm WARN deprecated stable@0.1.8: Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility
npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
npm WARN deprecated flatten@1.0.3: flatten is deprecated in favor of utility frameworks such as lodash.
npm WARN deprecated rollup-plugin-terser@7.0.2: This package has been deprecated and is no longer maintained. Please use @rollup/plugin-terser
npm WARN deprecated @types/localforage@0.0.34: This is a stub types definition for localforage (https://github.com/localForage/localForage). localforage provides its own type definitions, so you don't need @types/localforage installed!
npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
npm WARN deprecated sourcemap-codec@1.4.8: Please use @jridgewell/sourcemap-codec instead
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated browserslist@1.7.7: Browserslist 2 could fail on reading Browserslist >3.0 config used in other tools.
npm WARN deprecated trim@0.0.1: Use String.prototype.trim() instead
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated @thornbill/jellyfin-sdk@0.4.1: Use the @jellyfin/sdk package instead.
npm WARN deprecated core-js-pure@3.19.1: core-js-pure@<3.23.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Some versions have web compatibility issues. Please, upgrade your dependencies to the actual version of core-js-pure.
npm WARN deprecated core-js@3.20.2: core-js@<3.23.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Some versions have web compatibility issues. Please, upgrade your dependencies to the actual version of core-js.

> jellyfin-web@10.8.10 prepare
> node ./scripts/prepare.js


added 1761 packages in 3m
npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.

> jellyfin-web@10.8.10 build:production
> webpack --config webpack.prod.js

Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db

Why you should do it regularly:
https://github.com/browserslist/browserslist#browsers-data-updating
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
assets by path *.woff2 33.3 MiB 1103 assets
assets by path *.js 11.3 MiB 345 assets
assets by path assets/ 732 KiB 44 assets
assets by path *.png 708 KiB 32 assets
assets by path libraries/ 8.18 MiB 11 assets
assets by path themes/ 1.16 MiB 8 assets
assets by path *.json 2.65 KiB 3 assets
assets by path *.ts 19.7 KiB 2 assets
assets by path *.ico 13.3 KiB 2 assets
assets by path *.gif 3.99 KiB 2 assets
assets by path *.svg 2.2 KiB 2 assets
+ 5 assets
orphan modules 2.94 MiB [orphan] 997 modules
runtime modules 23 KiB 12 modules
modules by path ../node_modules/ 5.83 MiB (javascript) 33.8 MiB (asset)
  asset modules 45.5 KiB (javascript) 33.8 MiB (asset) 1109 modules
  javascript modules 5.79 MiB 629 modules
modules by path ./ 9.28 MiB (javascript) 28.6 KiB (asset) 512 modules
modules by mime type image/png 696 bytes
  data:image/png;base64,iVBORw0KGgoAAAAN.. 186 bytes [built] [code generated]
  data:image/png;base64,iVBORw0KGgoAAAAN.. 170 bytes [built] [code generated]
  + 2 modules
modules by mime type image/gif 218 bytes
  data:image/gif;base64,R0lGODlhCwAHAIAA.. 110 bytes [built] [code generated]
  data:image/gif;base64,R0lGODlhCwAHAIAA.. 108 bytes [built] [code generated]
+ 8 modules

LOG from InjectManifest
<i> The service worker at serviceworker.js will precache
<i>         1573 URLs, totaling 55.4 MB.

WARNING in libraries/subtitles-octopus-worker-legacy.js is 3.17 MB, and won't be precached. Configure maximumFileSizeToCacheInBytes to change this limit.

WARNING in asset size limit: The following asset(s) exceed the recommended size limit (244 KiB).
This can impact web performance.
Assets:
  main.jellyfin.bundle.js (1.2 MiB)
  194d1b9ef54d0014b3c4.ttf (311 KiB)
  themes/blueradiance/bg.jpg (933 KiB)
  libraries/subtitles-octopus-worker-legacy.js.mem (643 KiB)
  libraries/subtitles-octopus-worker.wasm (1.95 MiB)
  libraries/pdf.worker.js (984 KiB)
  libraries/subtitles-octopus-worker-legacy.js (3.02 MiB)
  libraries/subtitles-octopus-worker.js (354 KiB)
  libraries/wasm-gen/libarchive.wasm (875 KiB)
  4333.cfb1664db2e629f294eb.chunk.js (985 KiB)
  1891.8c2b31f7219aa2b4b8df.chunk.js (376 KiB)
  5487.a6e79037c4e3cc8353ff.chunk.js (334 KiB)
  8233.82228f339f6ae66a80d5.chunk.js (279 KiB)

WARNING in entrypoint size limit: The following entrypoint(s) combined asset size exceeds the recommended limit (244 KiB). This can impact web performance.
Entrypoints:
  main (1.2 MiB)
      main.jellyfin.bundle.js


webpack 5.70.0 compiled with 3 warnings in 115434 ms
PS E:\Tizen\gitted\jellyfin-web> cd..
PS E:\Tizen\gitted> cd jellyfin-tizen
>> $env:JELLYFIN_WEB_DIR="C:\jellyfin\jellyfin-web\dist"
>> npm ci --no-audit
npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.
npm WARN old lockfile
npm WARN old lockfile The package-lock.json file was created with an old version of npm,
npm WARN old lockfile so supplemental metadata must be fetched from the registry.
npm WARN old lockfile
npm WARN old lockfile This is a one-time fix-up, please be patient...
npm WARN old lockfile
npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
npm WARN deprecated request-promise-native@1.0.9: request-promise-native has been deprecated because it extends the now deprecated request package, see https://github.com/request/request/issues/3142
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
npm WARN deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependencies
npm WARN deprecated w3c-hr-time@1.0.2: Use your platform's native performance.now() and performance.timeOrigin.
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated gulp-util@3.0.8: gulp-util is deprecated - replace it, following the guidelines at https://medium.com/gulpjs/gulp-util-ca3b1f9f9ac5

> jellyfin-tizen@0.1.0 postinstall
> gulp

Using jellyfin-web from C:\jellyfin\jellyfin-web\dist
[02:28:46] Using gulpfile E:\Tizen\gitted\jellyfin-tizen\gulpfile.js
[02:28:46] Starting 'default'...
[02:28:46] Starting 'clean'...
[02:28:46] Finished 'clean' after 8.97 ms
[02:28:46] Starting 'searchFonts'...
[02:28:46] Finished 'searchFonts' after 12 ms
[02:28:46] Starting 'copy'...
[02:28:46] Starting 'modifyIndex'...
[02:28:46] Finished 'copy' after 6.49 ms
[02:28:46] 'modifyIndex' errored after 8.17 ms
[02:28:46] Error: File not found with singular glob: C:\jellyfin\jellyfin-web\dist/index.html (if this was purposeful, use `allowEmpty` option)
    at Glob.<anonymous> (E:\Tizen\gitted\jellyfin-tizen\node_modules\glob-stream\readable.js:84:17)
    at Object.onceWrapper (node:events:642:26)
    at Glob.emit (node:events:527:28)
    at Glob.emit (node:domain:475:12)
    at Glob._finish (E:\Tizen\gitted\jellyfin-tizen\node_modules\glob\glob.js:194:8)
    at done (E:\Tizen\gitted\jellyfin-tizen\node_modules\glob\glob.js:179:14)
    at Glob._processSimple2 (E:\Tizen\gitted\jellyfin-tizen\node_modules\glob\glob.js:685:12)
    at E:\Tizen\gitted\jellyfin-tizen\node_modules\glob\glob.js:673:10
    at Glob._stat2 (E:\Tizen\gitted\jellyfin-tizen\node_modules\glob\glob.js:769:12)
    at lstatcb_ (E:\Tizen\gitted\jellyfin-tizen\node_modules\glob\glob.js:761:12)
[02:28:46] 'default' errored after 41 ms
npm ERR! code 1
npm ERR! path E:\Tizen\gitted\jellyfin-tizen
npm ERR! command failed
npm ERR! command C:\WINDOWS\system32\cmd.exe /d /s /c gulp

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\**redacted**\AppData\Local\npm-cache\_logs\2023-05-24T06_28_00_229Z-debug-0.log
PS E:\Tizen\gitted\jellyfin-tizen>
```

**Comment by dmitrylyzo** [comment]
In your case
```ps
$env:JELLYFIN_WEB_DIR="E:\Tizen\gitted\jellyfin-web\dist"
```

**Comment by WavesOfSalt** [comment]
The universal one worked fine. Damn. I just don't read I guess. But the Build WGT command doesn't work because apparently no such bat or exe exists, despite having Tizen Studio installed.

**Comment by dmitrylyzo** [comment]
> But the Build WGT command doesn't work because apparently no such bat or exe exists, despite having Tizen Studio installed.

Because **all tools (node, tizen) are assumed to be added to the PATH environment variable** (what is stated in the guide), but by default they are not.
I thought about adding the full path to the commands (or maybe add a command to set the PATH environment variable), but haven't decided yet.

`tizen.bat` is usually located at `c:\tizen-studio\tools\ide\bin\tizen.bat`.

**Comment by WavesOfSalt** [comment]
Yo! I didn't mean to leave this open. Around the same time, I found someone that provided this precompiled. For some reason, my TV would just accept anything sent to it without any error message. Couldn't even install things from the Samsung store despite having enough storage. I just stuck to the web browser, since it appears to support all codecs, including AV1, just only through Jellyfin. Anyways, I gave up on attempting to build my own because of that message-free bug and my tyzen certificate (?) expired.
